### PR TITLE
Schedule adding of calendars

### LINKED ISSUE #92: Add calendars for new classes when they are created
A calendar should be created and added to the database for every new class that is created when an academic year starts

### PR DESCRIPTION
**Pull request by DenverCoder1**
Closes #92 

A calendar should be created and added to the database for every new class that is created when an academic year starts.

### CODE REVIEW

**Comment by DenverCoder1** [comment]
@abrahammurciano For simplicity, the `get_campuses` part and the `Campus` class are basically copied from `create_class`. Maybe you have a better idea for doing that?

**Comment by abrahammurciano** [comment]
They should probably be moved to utils then. `/utils/campus/campus.py` and `/utils/campus/queries/get_campuses.sql`.
The function get_campuses can be a static method of the campus class.

File: modules/calendar/calendar_creator.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+	async def __get_campuses(self) -> Iterable[Campus]:
+		"""Fetch a list of campuses from the database"""
+		query = self.__sql_fetcher["get_campuses.sql"]
+		with self.__conn as conn:
+			with conn.cursor() as cursor:
+				cursor.execute(query)
+				campuses = [Campus(*tup) for tup in cursor.fetchall()]
+		return campuses
+
+	async def __add_calendar(self, calendar_id: str, campus_id: int, year: int):
```
Comment (abrahammurciano) [inline modules/calendar/calendar_creator.py:33]:
```suggestion
	def __add_calendar(self, calendar_id: str, campus_id: int, year: int):
```
You're not awaiting anything, no need to be async

File: modules/calendar/calendar_creator.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+	async def create_class_calendars(self, year: int):
+		"""Create a calendar for each campus and add it to the database"""
+		for campus in await self.__get_campuses():
+			# create calendar
+			calendar = self.__service.create_calendar(f"JCT CompSci {campus.name} {year}")
+			# update class in database
+			await self.__add_calendar(calendar.id, campus.id, year)
+
+	async def __get_campuses(self) -> Iterable[Campus]:
```
Comment (abrahammurciano) [inline modules/calendar/calendar_creator.py:24]: To be moved to utils, no need to be async

File: modules/calendar/calendar_creator.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+class CalendarCreator:
+	def __init__(
+		self, service: CalendarService, conn: sql.connection, sql_fetcher: SqlFetcher
+	):
+		self.__conn = conn
+		self.__sql_fetcher = sql_fetcher
+		self.__service = service
+
+	async def create_class_calendars(self, year: int):
```
Comment (abrahammurciano) [inline modules/calendar/calendar_creator.py:16]:
```suggestion
	def create_class_calendars(self, year: int):
```
No need for async

File: modules/calendar/calendar_creator.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+		"""Create a calendar for each campus and add it to the database"""
+		for campus in await self.__get_campuses():
+			# create calendar
+			calendar = self.__service.create_calendar(f"JCT CompSci {campus.name} {year}")
+			# update class in database
+			await self.__add_calendar(calendar.id, campus.id, year)
```
Comment (abrahammurciano) [inline modules/calendar/calendar_creator.py:22]:
```suggestion
			await self.__add_calendar(calendar.id, campus.id, year)
```
This function doesn't need to be a coroutine

File: modules/calendar/campus.py [hunk 5]
```diff
@@ -0,0 +1,4 @@
+class Campus:
```
Comment (abrahammurciano) [inline modules/calendar/campus.py:1]: To be moved to `/utils/campus/campus.py`.

`get_campuses` should be a public static method of this class

File: modules/calendar/queries/get_campuses.sql [hunk 6]
```diff
@@ -0,0 +1 @@
+SELECT id, name FROM campuses
```
Comment (abrahammurciano) [inline modules/calendar/queries/get_campuses.sql:1]: To be moved to `/utils/campus/queries/get_campuses.sql`

**Review by abrahammurciano — CHANGES_REQUESTED** [review]
Well done

**Review by abrahammurciano — APPROVED** [review]

**Pull request merged**